core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -230,27 +232,20 @@ SnapshotSummary.Builder buildSummary(Iterable<ManifestFile> manifests) {
    *
    * @param manifests a set of filtered manifests
    */
-  @SuppressWarnings("CollectionUndefinedEquality")
   private void validateRequiredDeletes(ManifestFile... manifests) {
     if (failMissingDeletePaths) {
-      CharSequenceSet deletedFiles = deletedFiles(manifests);
-      ValidationException.check(
-          deletedFiles.containsAll(deletePaths),
-          "Missing required files to delete: %s",
-          COMMA.join(Iterables.filter(deletePaths, path -> !deletedFiles.contains(path))));
+      deletedFiles(manifests).validateRequiredDeletes(filesToDelete);
```

Comment:
What if we have a path-only delete that is missing?

Reply:
good catch, I was relying on existing tests to catch this but after your comment I realized that there's no test coverage for this available. I added test coverage and fixed this